src/utils/metrics/reporter.go

```diff
@@ -0,0 +1,120 @@
+package metrics
+
+import (
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"go.uber.org/zap"
+)
+
+// Reporter gathers metrics across jobs and reports them.
+// Concurrency-safe.
+type Reporter interface {
+	// WriteSummary dumps Reporter contents into the target.
+	WriteSummary()
+
+	SumBytesSent() uint64
+	ClearData()
+
+	// NewAccumulator returns a new metrics Accumulator for the Reporter.
+	NewAccumulator(jobID string) *Accumulator
+}
+
+// ZapReporter
+
+type ZapReporter struct {
+	data   ReporterData
```

Comment:
You can use type embedding to avoid specifying methods that just call the same method of data

Reply:
after complete separation of metrics data from reporters I didn't even have to, and the code became shorter. Well, at least I learned what those type embeddings are